### ISSUE TITLE
C# unit test failure

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#5269]
Something I did today apparently caused the following:

```
Command: haxe [compile-cs-travis.hxml,-D,erase_generics,-D,fast_cast]
haxelib run hxcs hxcs_build.txt --haxe-version 3300 --feature-level 1
/home/travis/build/HaxeFoundation/haxe/tests/unit/src/unit/TestCSharp.hx(314,81): error CS0030: Cannot convert type `null' to `haxe.test.TEnumWithFlag'
/home/travis/build/HaxeFoundation/haxe/tests/unit/src/unit/TestCSharp.hx(337,87): error CS0030: Cannot convert type `null' to `haxe.test.TEnumWithBigFlag'
Compilation failed: 2 error(s), 0 warnings
Compilation error
Native compilation failed
```

**Comment by Simn** [comment]
Yeah I'm gonna need some help on this one, must be something very specific.

**Comment by nadako** [comment]
Minimal example:
```haxe
abstract Flags<T>(T) {
    public inline function new(?initial:T) {
		this = initial;
    }
}

@:nativeGen
enum E {
    A;B;C;
}

class Main {
    static function main() {
        var f = new Flags<E>();
        trace(f);
    }
}
```

C# with `-D erase-generics`
```cs
global::E this1 = ((global::E) (null) );
```

Without `-D erase-generics` it generates this:
```cs
global::E this1 = (default(global::haxe.lang.Null<global::E>)).@value;
```

**Comment by waneck** [comment]
Fixed by ae36c5696150b6349c8faef3c4dd9454e70a0a77
